controller/node-js/server/remote-keyboard.js

```diff
@@ -10,6 +10,17 @@
 function RemoteKeyboard (commandHandler) {
   const pressedKeys = new Set()
 
+  // This deconstruction does not work. The functions are not called. Why?
```

Comment:
What is this supposed to do?

Reply:
This is not really critical, but in JS it is common to assign object properties to local variables, like this:
```
const {reset, goForward, goBackward, forwardLeft, forwardRight...} = commandHandler
```
is equivalent to:
```
const reset = commandHandler.reset
```
So, instead of calling:```commandHandler.reset()```we can just call:```reset()```

But in this case, if we call ```reset()```, the function is not called. I was wondering why, because functions are first-class citizens in JS, and should be able to use them just like any other property. Maybe somebody can answer.

Reply:
Check this. 
https://stackoverflow.com/questions/45910018/how-to-bind-methods-when-destructuring-an-object-in-javascript